withRenew.js

```diff
@@ -45,44 +45,49 @@ const withRenew = WrappedComponent =>
     }
 
     renew(loan, patron, bulkRenewal) {
+      this.setState({ bulkRenewal });
       const params = {
         itemBarcode: loan.item.barcode,
         userBarcode: patron.barcode,
       };
 
-      this.setState({ bulkRenewal });
-
-      return this.props.mutator.renew.POST(params).catch(resp => {
-        const contentType = resp.headers.get('Content-Type');
-        if (contentType && contentType.startsWith('application/json')) {
-          resp.json().then(error => this.handleErrors(error));
-        } else {
-          // eslint-disable-next-line no-alert
-          resp.text().then(error => alert(error));
-        }
-        throw new Error(resp);
+      return new Promise((resolve, reject) => {
+        this.props.mutator.renew.POST(params).then(resolve).catch(resp => {
+          const contentType = resp.headers.get('Content-Type');
+          if (contentType && contentType.startsWith('application/json')) {
+            resp.json().then((error) => {
+              const errors = this.handleErrors(error);
+              reject(this.getMessage(errors));
+            });
+          } else {
+            resp.text().then((error) => {
+              reject(error);
+              alert(error); // eslint-disable-line no-alert
+            });
+          }
+        });
       });
     }
 
     handleErrors(error) {
       const errors = (error.errors || []).map(err => err.message);
       this.setState({ errors });
+      return errors;
     }
 
-
     hideModal() {
       this.setState({ errors: [] });
     }
 
-    getMessage() {
-      const { errors } = this.state;
-      if (!errors.length) return '';
-      return errors.reduce((msg, error) => ((msg) ? `${msg}, ${error}` : error), '');
+    getMessage(errors) {
+      this.errors = errors;
```

Comment:
@aditya-matukumalli let's avoid creating an instance var here. The lint will complain about `getMessage` but we can add ignore or just move getMessage out of the component.

Reply:
Sure, will do